packages/rocketchat-livechat/server/startup.js

```diff
@@ -11,6 +11,19 @@ Meteor.startup(() => {
 		return room.t === 'l' && extraData && extraData.token && room.v && room.v.token === extraData.token;
 	});
 
+	RocketChat.authz.addRoomAccessValidator(function(room, file) {
```

Comment:
any way to use these new `addRoomAccessValidator` checks to allow viewing files from without having to [change the file URLs](https://github.com/RocketChat/Rocket.Chat/pull/10514/files#diff-907472ebc361e1ad91b8ec4fd78b1b2fR4) to `/livechat-file-upload/`? I give this a look.

Reply:
I added the validation using `addRoomAccessValidator` in the `requests.js` file.
Since we're using this same method in `FileUploadBase.js`, it seems that it makes sense tu use the same method to check if the Livechat visitor is allowed to access the file.